<launch>
    <arg name="robot_name" default="thorvald_001"/>
    <arg name="tf_prefix" default="$(arg robot_name)/"/>
    <arg name="tmap" default="topological_map" />

    <!--  topo_nav.launch from uol_cmp9767m_tutorials -->
    <include file="topo_nav.launch" >
        <arg name="move_base_reconf_service" default="DWAPlannerROS" />
        <arg name="tmap" value="$(arg tmap)" />
    </include>

    <group ns="$(arg robot_name)">
        <remap from="map" to="/map" />

        <!-- launch robot_pose_publisher -->
        <node pkg="robot_pose_publisher" type="robot_pose_publisher" name="robot_pose_publisher" respawn="true">
            <param name="~map_frame" value="map"/>
            <param name="~base_frame" value="$(arg tf_prefix)base_link"/>
        </node>

        <!-- move_base -->
        <node name="move_base" pkg="move_base" type="move_base" output="screen">
            <remap from="odom" to="odometry/base_raw" />
            <remap from="cmd_vel" to="nav_vel" />

            <!-- planner config -->
            <rosparam file="$(find thorvald_2dnav)/config/base_local_planner_params.yaml" command="load"/>
            <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
            <!-- <param name="base_local_planner" value="dwa_local_planner/TrajectoryPlannerROS" /> -->

            <!-- costmap config -->
            <rosparam file="$(find thorvald_2dnav)/config/costmap_common_params.yaml" command="load" ns="global_costmap"/> 
            <rosparam file="$(find thorvald_2dnav)/config/costmap_common_params.yaml" command="load" ns="local_costmap"/>
            <rosparam file="$(find thorvald_2dnav)/config/global_costmap_params.yaml" command="load"/> 
            <rosparam file="$(find thorvald_2dnav)/config/local_costmap_params.yaml" command="load"/>

            <!--remap hokuyo frame and topic-->
            <rosparam param="local_costmap/obstacle_layer/laser_scan_sensor/sensor_frame" subst_value="True">$(arg tf_prefix)hokuyo</rosparam>
            <rosparam param="local_costmap/obstacle_layer/laser_scan_sensor/topic" subst_value="True">scan</rosparam>
            <rosparam param="global_costmap/global_obstacle_layer/laser_scan_sensor/sensor_frame" subst_value="True">$(arg tf_prefix)hokuyo</rosparam>
            <rosparam param="global_costmap/global_obstacle_layer/laser_scan_sensor/topic" subst_value="True">scan</rosparam>
        </node>

        
    </group>

</launch>